lib/Parse/ParseType.cpp

```diff
@@ -1488,22 +1486,21 @@ bool Parser::canParseAsGenericArgumentList() {
 
 bool Parser::canParseGenericArguments() {
   // Parse the opening '<'.
-  if (!startsWithLess(Tok))
-    return false;
-  consumeStartingLess();
-  
-  do {
-    if (!canParseType())
-      return false;
-    // Parse the comma, if the list continues.
-  } while (consumeIf(tok::comma));
-  
-  if (!startsWithGreater(Tok)) {
-    return false;
-  } else {
-    consumeStartingGreater();
-    return true;
+  if (startsWithLess(Tok)) {
```

Comment:
As we've discussed in the past, it is intentional to prefer early return. I think this should restore the original `if (!startsWithLess(Tok)) return false;`.